SocketClientBraitenburg.py

```diff
@@ -0,0 +1,316 @@
+#!/usr/bin/env python3
+from ev3dev.ev3 import *
+from math import sin, cos
+from time import sleep, time
+from writeCSV import WriteCSV
+import pickle
+import threading
+from Ev3devSetup import Ev3devSetup
+import socket
+
+# ============================================
+# A thread class from https://www.tutorialspoint.com/python/python_multithreading.htm
+# This thread class represents a background thread on the robot to collect sensor-motor data and send it back
+# to the server.
+class SensorBackgroundThread (threading.Thread):
+    def __init__(self, threadID, name, counter):
+        threading.Thread.__init__(self)
+        self.threadID = threadID
+        self.name = name
+        self.counter = counter
+    def run(self):
+        while True:
+            sensorValues(self.name)
+
+# =============================================
+
+# commented this out to reduce annoying beep
+# Sound.beep()
+
+MAX_SENSOR = 100.0  # percent
+MAX_MOTOR = 1000.0
+
+BIAS = 0.05
+SENSOR_GAIN = 1.0
+OUTPUT_GAIN = 1.0
+
+Leds.set_color(Leds.LEFT, Leds.YELLOW)
+Leds.set_color(Leds.RIGHT, Leds.YELLOW)
+
+"""
+lm = LargeMotor('outB')
+rm = LargeMotor('outC')
+lm.reset()
+rm.reset()
+b = Button()
+
+ls = ColorSensor('in2')
+# assert ls.connected, "Left sensor not connected to port 2"
+rs = ColorSensor('in3')
+# assert rs.connected, "Right sensor not connected to port 3"
+ls.mode = 'COL-AMBIENT'
+rs.mode = 'COL-AMBIENT'
+
+lu = UltrasonicSensor('in1')
+lu.mode = 'US-DIST-CM'
+ru = UltrasonicSensor('in4')
+ru.mode = 'US-DIST-CM'
+"""
+
+#Setting up with MAX_SENSOR, MAX_MOTOR, BIAS, SENSOR_GAIN, OUTPUT_GAIN
+ev3devrobot = Ev3devSetup()
+
+motor_left = ev3devrobot.initLargeMotor('outB')
+motor_left.reset()
+
+motor_right = ev3devrobot.initLargeMotor('outC')
+motor_right.reset()
+
+left_colour_sensor = ev3devrobot.initColorSensor('in2')
+left_colour_sensor.mode = 'COL-AMBIENT'
+
+right_colour_sensor = ev3devrobot.initColorSensor('in3')
+right_colour_sensor.mode = 'COL-AMBIENT'
+
+left_ultrasonic_sensor = ev3devrobot.initUltraSonicSensor('in1')
+left_ultrasonic_sensor.mode = 'US-DIST-CM'
+
+right_ultrasonic_sensor = ev3devrobot.initUltraSonicSensor('in4')
+right_ultrasonic_sensor.mode = 'US-DIST-CM'
+
+# ===============================================
+
+def cleanup():
+    print('cleaning up...')
+    motor_left.stop()
+    motor_right.stop()
+    print('quiting')
+    exit()
+
+
+btn = Button()
+
+# ===============================================
+
+# Do something when state of any button changes:
+def right(state):  # neater use of 'if' follows:
+    global SENSOR_GAIN
+    if state:
+        SENSOR_GAIN += 0.05
+        print('SENSOR_GAIN:', SENSOR_GAIN)
+
+# ===============================================
+
+def left(state):
+    global SENSOR_GAIN
```

Comment:
is this the same SENSOR_GAIN set up in ev3devsetup?

Reply:
Yup

Reply:
I just fused the cs765 and the SocketClient together - thereby not being efficient